testplan/testing/base.py

```diff
@@ -862,13 +867,83 @@ def _record_driver_timing(
         case_report.extend(case_result.serialized_entries)
         case_report.attachments.extend(case_result.attachments)
 
+    def _record_driver_connection(self, case_report: TestCaseReport) -> None:
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        connections: list[BaseDriverConnection] = []
+        for driver in self.resources:
+            try:
+                for conn_info in driver.extract_driver_metadata().conn_info:
+                    connection = conn_info.connectionType
+                    if issubclass(connection, BaseDriverConnection):
+                        added = False
+                        for existing_connection in connections:
+                            added = existing_connection.add_driver_if_in_connection(
+                                str(driver), conn_info
+                            )
+                            if added:
+                                break
+                        if not added:
+                            new_connection = connection(conn_info)
```

Comment:
i don't quite like this idea here, suggest associate those classes through methods, maybe `new_connection = conn_info.promote_to_connection()`